examples/knative/templates/server.ps1

```diff
@@ -1,141 +1,156 @@
-      exit 1
-   }
-
-   while($true) {
-      $breakSignal = Start-HttpCloudEventListener -Url $url
-      if ($breakSignal) {
-         Write-Host "$(Get-Date) - PowerShell HTTP server stop requested"
-         break;
-      }
-   }
+        }
+        catch {
+            Write-Error "$(Get-Date) - Listener Processing Error: $($_.Exception.ToString())"
+            exit 1
+        }
+        finally {
+            $listener.Stop()
+        }
+    }
+
+    # Runs Init function (defined in handler.ps1) which can be used to enable warm startup
+    try {
+        Process-Init
+    }
+    catch {
+        Write-Error "$(Get-Date) - Init Processing Error: $($_.Exception.ToString())"
+        exit 1
+    }
+
+    $breakSignal = Start-HttpCloudEventListener -Url $url
+    if ($breakSignal) {
+        Write-Host "$(Get-Date) - PowerShell HTTP server stop requested"
+        break;
+    }
 } -ArgumentList $url, $serverStopMessage
 
 $killEvent = new-object 'System.Threading.AutoResetEvent' -ArgumentList $false
 
 $serverTerminateJob = Start-ThreadJob {
-param($killEvent, $url, $serverStopMessage)
-   $killEvent.WaitOne() | Out-Null
-   Invoke-WebRequest -Uri $url -Body $serverStopMessage | Out-Null
+    param($killEvent, $url, $serverStopMessage)
+    $killEvent.WaitOne() | Out-Null
+    Invoke-WebRequest -Uri $url -Body $serverStopMessage | Out-Null
 } -ArgumentList $killEvent, $localUrl, $serverStopMessage
 
 try {
-   Write-Host "$(Get-Date) - PowerShell HTTP server start listening on '$url'"
-   $running = $true
-   while($running) {
-      Start-Sleep  -Milliseconds 100
-      $running = ($backgroundServer.State -eq 'Running')
-      $backgroundServer = $backgroundServer | Get-Job
-      $backgroundServer | Receive-Job
-   }
-} finally {
-   Write-Host "$(Get-Date) - PowerShell HTTP Server stop requested. Waiting for server to stop"
-   $killEvent.Set() | Out-Null
-   Get-Job | Wait-Job | Receive-Job
-   Write-Host "$(Get-Date) - PowerShell HTTP server is stopped"
-
-   # Runs Shutdown function (defined in handler.ps1) to clean up connections from warm startup
-   try {
-      Process-Shutdown
-   } catch {
-      Write-Error "`n$(Get-Date) - Shutdown Processing Error: $($_.Exception.ToString())"
-   }
+    Write-Host "$(Get-Date) - PowerShell HTTP server start listening on '$url'"
+    $running = $true
+    while ($running) {
+        Start-Sleep  -Milliseconds 100
+        $running = ($backgroundServer.State -eq 'Running')
+        $backgroundServer = $backgroundServer | Get-Job
+        $backgroundServer | Receive-Job
+    }
+}
+finally {
+    Write-Host "$(Get-Date) - PowerShell HTTP Server stop requested. Waiting for server to stop"
+    $killEvent.Set() | Out-Null
+    Get-Job | Wait-Job | Receive-Job
+    Write-Host "$(Get-Date) - PowerShell HTTP server is stopped"
+
+    # Runs Shutdown function (defined in handler.ps1) to clean up connections from warm startup
+    try {
+        Process-Shutdown
```

Comment:
Should we run this before stopping the server? Just thinking about logic flow here, where I'd expect the server to be the last resource to be stopped after processing all hooks.